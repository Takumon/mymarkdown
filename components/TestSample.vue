<template>
  <ul>
    <li v-for="(item, index) in list" :key="index">
      {{ item }}
    </li>
  </ul>
</template>

<script>
  export default {
    name: 'testSample',
    props: {
      list: {
        type: Array,
        required: true,
      },
    },
  }
</script>