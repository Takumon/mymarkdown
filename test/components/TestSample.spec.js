import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import TestSample from '../../components/TestSample.vue'


const localVue = createLocalVue()
localVue.use(Vuex)


describe('TestSample.js', () => {
  let wrapper
  let store
  const list = ['Coffee', 'Tea', 'Snicker', 'Mars']

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        showLoading: false,
      },
      actions: {
        setShowLoading ({ commit }, isShow) {
          return new Promise((resolve, reject) => {
            commit( 'setShowLoading', { isShow })
            resolve()
          })
        },
      },
      mutations: {
        setShowLoading (state, payload) {
          state.showLoading = payload.isShow
        },
      }
    })

    wrapper = shallowMount(TestSample, {
      propsData: { list },
      store,
      localVue
    })
  })
  describe('初期表示', () => {

    test('初期表示スナップショット', () => {
      expect({ html: wrapper.html() }).toMatchSnapshot()
    })

    test('初期表示 リストが表示される', () => {
      const $items = wrapper.findAll('li').wrappers

      $items.forEach(($item, index) => {
        expect($item.text()).toEqual(list[index])
      })
    })

    test('初期表示 ボタンがfalseで表示される', () => {
      const $button = wrapper.find('#toggleLoading')
      expect($button.text()).toEqual('false')
    })
  })


  describe('ボタンクリック後', () => {

    test('ボタンがtrueで表示される', () => {
      const $button = wrapper.find('#toggleLoading')
      $button.trigger('click')
      expect($button.text()).toEqual('true')
    })
  })
})
