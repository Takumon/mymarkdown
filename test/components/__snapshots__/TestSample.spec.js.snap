// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSample.js 初期表示 初期表示スナップショット 1`] = `
Object {
  "html": "<div><ul><li>
      Coffee
    </li><li>
      Tea
    </li><li>
      Snicker
    </li><li>
      Mars
    </li></ul> <button id=\\"toggleLoading\\">false</button></div>",
}
`;
